File: java/WebForms.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class WebForms {

    private Map<String, String> webFormsData = new HashMap<>();

    // Add
    public void addId(String inputPlace, String id) {
        webFormsData.put("ai" + inputPlace, id);
    }

    public void addName(String inputPlace, String name) {
        webFormsData.put("an" + inputPlace, name);
    }

    public void addValue(String inputPlace, String value) {
        webFormsData.put("av" + inputPlace, value);
    }

    public void addClass(String inputPlace, String className) {
        webFormsData.put("ac" + inputPlace, className);
    }

    public void addStyle(String inputPlace, String style) {
        webFormsData.put("as" + inputPlace, style);
    }

    public void addOptionTag(String inputPlace, String text, String value, boolean selected) {
        webFormsData.put("ao" + inputPlace, value + '|' + text + (selected ? "|1" : ""));
    }

    public void addCheckBoxTag(String inputPlace, String text, String value, boolean checked) {
        webFormsData.put("ak" + inputPlace, value + '|' + text + (checked ? "|1" : ""));
    }

    public void addTitle(String inputPlace, String title) {
        webFormsData.put("al" + inputPlace, title);
    }

    public void addText(String inputPlace, String text) {
        webFormsData.put("at" + inputPlace, text.replace("\n", "$[ln];"));
    }

    public void addAttribute(String inputPlace, String attribute, String value) {
        webFormsData.put("aa" + inputPlace, attribute + '|' + value);
    }

    public void addTag(String inputPlace, String tagName, String id) {
        webFormsData.put("nt" + inputPlace, tagName + (!id.isEmpty() ? '|' + id : ""));
    }

    // Set
    public void setId(String inputPlace, String id) {
        webFormsData.put("si" + inputPlace, id);
    }

    public void setName(String inputPlace, String name) {
        webFormsData.put("sn" + inputPlace, name);
    }

    public void setValue(String inputPlace, String value) {
        webFormsData.put("sv" + inputPlace, value);
    }

    public void setClass(String inputPlace, String className) {
        webFormsData.put("sc" + inputPlace, className);
    }

    public void setStyle(String inputPlace, String style) {
        webFormsData.put("ss" + inputPlace, style);
    }

    public void setOptionTag(String inputPlace, String text, String value, boolean selected) {
        webFormsData.put("so" + inputPlace, value + '|' + text + (selected ? "|1" : ""));
    }

    public void setChecked(String inputPlace, boolean checked) {
        webFormsData.put("sk" + inputPlace, checked ? "1" : "0");
    }

    public void setCheckBoxTagToList(String inputPlace, String text, String value, boolean checked) {
        webFormsData.put("sk" + inputPlace, value + '|' + text + (checked ? "|1" : ""));
    }

    public void setTitle(String inputPlace, String title) {
        webFormsData.put("sl" + inputPlace, title);
    }

    public void setText(String inputPlace, String text) {
        webFormsData.put("st" + inputPlace, text.replace("\n", "$[ln];"));
    }

    public void setAttribute(String inputPlace, String attribute, String value) {
        webFormsData.put("sa" + inputPlace, attribute + (!value.isEmpty() ? '|' + value : ""));
    }

    public void setWidth(String inputPlace, String width) {
        webFormsData.put("sw" + inputPlace, width);
    }

    public void setWidth(String inputPlace, int width) {
        setWidth(inputPlace, width + "px");
    }

    public void setHeight(String inputPlace, String height) {
        webFormsData.put("sh" + inputPlace, height);
    }

    public void setHeight(String inputPlace, int height) {
        setHeight(inputPlace, height + "px");
    }

    // Insert
    public void insertId(String inputPlace, String id) {
        webFormsData.put("ii" + inputPlace, id);
    }

    public void insertName(String inputPlace, String name) {
        webFormsData.put("in" + inputPlace, name);
    }

    public void insertValue(String inputPlace, String value) {
        webFormsData.put("iv" + inputPlace, value);
    }

    public void insertClass(String inputPlace, String className) {
        webFormsData.put("ic" + inputPlace, className);
    }

    public void insertStyle(String inputPlace, String style) {
        webFormsData.put("is" + inputPlace, style);
    }

    public void insertOptionTag(String inputPlace, String text, String value, boolean selected) {
        webFormsData.put("io" + inputPlace, value + '|' + text + (selected ? "|1" : ""));
    }

    public void insertCheckBoxTag(String inputPlace, String text, String value, boolean checked) {
        webFormsData.put("ik" + inputPlace, value + '|' + text + (checked ? "|1" : ""));
    }

    public void insertTitle(String inputPlace, String title) {
        webFormsData.put("il" + inputPlace, title);
    }

    public void insertText(String inputPlace, String text) {
        webFormsData.put("it" + inputPlace, text.replace("\n", "$[ln];"));
    }

    public void insertAttribute(String inputPlace, String attribute, String value) {
        webFormsData.put("ia" + inputPlace, attribute + (!value.isEmpty() ? '|' + value : ""));
    }

    // Delete
    public void deleteId(String inputPlace) {
        webFormsData.put("di" + inputPlace, "1");
    }

    public void deleteName(String inputPlace) {
        webFormsData.put("dn" + inputPlace, "1");
    }

    public void deleteValue(String inputPlace) {
        webFormsData.put("dv" + inputPlace, "1");
    }

    public void deleteClass(String inputPlace, String className) {
        webFormsData.put("dc" + inputPlace, className);
    }

    public void deleteStyle(String inputPlace, String styleName) {
        webFormsData.put("ds" + inputPlace, styleName);
    }

    public void deleteOptionTag(String inputPlace, String value) {
        webFormsData.put("do" + inputPlace, value);
    }

    public void deleteCheckBoxTag(String inputPlace, String value) {
        webFormsData.put("dk" + inputPlace, value);
    }

    public void deleteTitle(String inputPlace) {
        webFormsData.put("dl" + inputPlace, "1");
    }

    public void deleteText(String inputPlace) {
        webFormsData.put("dt" + inputPlace, "1");
    }

    public void deleteAttribute(String inputPlace, String attribute) {
        webFormsData.put("da" + inputPlace, attribute);
    }

    public void delete(String inputPlace) {
        webFormsData.put("de" + inputPlace, "1");
    }

    // Other
    public void setBackgroundColor(String inputPlace, String color) {
        webFormsData.put("bc" + inputPlace, color);
    }

    public void setTextColor(String inputPlace, String color) {
        webFormsData.put("tc" + inputPlace, color);
    }

    public void setFontName(String inputPlace, String name) {
        webFormsData.put("fn" + inputPlace, name);
    }

    public void setFontSize(String inputPlace, String size) {
        webFormsData.put("fs" + inputPlace, size);
    }

    public void setFontSize(String inputPlace, int size) {
        webFormsData.put("fs" + inputPlace, size + "px");
    }

    public void setFontBold(String inputPlace, boolean bold) {
        webFormsData.put("fb" + inputPlace, bold ? "1" : "0");
    }

    public void setVisible(String inputPlace, boolean visible) {
        webFormsData.put("vi" + inputPlace, visible ? "1" : "0");
    }

    public void setTextAlign(String inputPlace, String align) {
        webFormsData.put("ta" + inputPlace, align);
    }

    public void setReadOnly(String inputPlace, boolean readOnly) {
        webFormsData.put("sr" + inputPlace, readOnly ? "1" : "0");
    }

    public void setDisabled(String inputPlace, boolean disabled) {
        webFormsData.put("sd" + inputPlace, disabled ? "1" : "0");
    }

    public void setMinLength(String inputPlace, int length) {
        webFormsData.put("mn" + inputPlace, Integer.toString(length));
    }

    public void setMaxLength(String inputPlace, int length) {
        webFormsData.put("mx" + inputPlace, Integer.toString(length));
    }

    public void setSelectedValue(String inputPlace, String value) {
        webFormsData.put("ts" + inputPlace, value);
    }

    public void setSelectedIndex(String inputPlace, int index) {
        webFormsData.put("ti" + inputPlace, Integer.toString(index));
    }

    public void setCheckedValue(String inputPlace, String value, boolean selected) {
        webFormsData.put("ks" + inputPlace, value + "|" + (selected ? "1" : "0"));
    }

    public void setCheckedIndex(String inputPlace, int index, boolean selected) {
        webFormsData.put("ki" + inputPlace, index + "|" + (selected ? "1" : "0"));
    }

    public void callScript(String scriptText) {
        webFormsData.put("_", scriptText.replace("\n", "$[ln];"));
    }

    public void loadUrl(String inputPlace, String url) {
        webFormsData.put("lu" + inputPlace, url);
    }

    // Increase
    public void increaseMinLength(String inputPlace, int value) {
        webFormsData.put("+n" + inputPlace, Integer.toString(value));
    }

    public void increaseMaxLength(String inputPlace, int value) {
        webFormsData.put("+x" + inputPlace, Integer.toString(value));
    }

    public void increaseFontSize(String inputPlace, int value) {
        webFormsData.put("+f" + inputPlace, Integer.toString(value));
    }

    public void increaseWidth(String inputPlace, int value) {
        webFormsData.put("+w" + inputPlace, Integer.toString(value));
    }

    public void increaseHeight(String inputPlace, int value) {
        webFormsData.put("+h" + inputPlace, Integer.toString(value));
    }

    public void increaseValue(String inputPlace, int value) {
        webFormsData.put("+v" + inputPlace, Integer.toString(value));
    }

    // Decrease
    public void decreaseMinLength(String inputPlace, int value) {
        webFormsData.put("-n" + inputPlace, Integer.toString(value));
    }

    public void decreaseMaxLength(String inputPlace, int value) {
        webFormsData.put("-x" + inputPlace, Integer.toString(value));
    }

    public void decreaseFontSize(String inputPlace, int value) {
        webFormsData.put("-f" + inputPlace, Integer.toString(value));
    }

    public void decreaseWidth(String inputPlace, int value) {
        webFormsData.put("-w" + inputPlace, Integer.toString(value));
    }

    public void decreaseHeight(String inputPlace, int value) {
        webFormsData.put("-h" + inputPlace, Integer.toString(value));
    }

    public void decreaseValue(String inputPlace, int value) {
        webFormsData.put("-v" + inputPlace, Integer.toString(value));
    }

    // Pre Runner
    public void assignDelay(float second, int index) {
        String currentName = getNameByIndex(index);

        if (currentName.isEmpty()) {
            return;
        }

        changeNameByIndex(index, ":" + second + ")" + currentName);
    }

    public void assignDelayChange(float second, int index) {
        String currentName = getNameByIndex(index);

        if (currentName.isEmpty()) {
            return;
        }

        currentName = removeOuter(currentName, ":", ")");

        changeNameByIndex(index, ":" + second + ")" + currentName);
    }

    public void assignInterval(float second, int index) {
        String currentName = getNameByIndex(index);

        if (currentName.isEmpty()) {
            return;
        }

        changeNameByIndex(index, "(" + second + ")" + currentName);
    }

    public void assignIntervalChange(float second, int index) {
        String currentName = getNameByIndex(index);

        if (currentName.isEmpty()) {
            return;
        }

        currentName = removeOuter(currentName, "(", ")");

        changeNameByIndex(index, "(" + second + ")" + currentName);
    }

    // Get
    public String getFormsActionData() {
        StringBuilder returnValue = new StringBuilder();

        for (Map.Entry<String, String> entry : webFormsData.entrySet()) {
            returnValue.append(System.lineSeparator()).append(entry.getKey()).append('=').append(entry.getValue());
        }

        return returnValue.toString();
    }

    public String response() {
        return "[web-forms]" + getFormsActionData();
    }

    public String getFormsActionDataLineBreak() {
        StringBuilder returnValue = new StringBuilder();
        List<String> webFormsDataList = new ArrayList<>(webFormsData.keySet());

        int i = webFormsDataList.size();
        for (String key : webFormsDataList) {
            returnValue.append(key)
                       .append('=')
                       .append(webFormsData.get(key).replace("\"", "$[dq];"))
                       .append((i-- > 1) ? "$[sln];" : "");
        }

        return returnValue.toString();
    }

    // Export
    public String exportToWebFormsTag(String src) {
        return "<web-forms ac=\"" + getFormsActionDataLineBreak() + "\"" + (!src.isEmpty() ? " src=\"" + src + "\"" : "") + "></web-forms>";
    }

    // Overload
    public String exportToWebFormsTag(String width, String height, String src) {
        return "<web-forms ac=\"" + getFormsActionDataLineBreak() + "\" width=\"" + width + "\" height=\"" + height + "\"" + (!src.isEmpty() ? " src=\"" + src + "\"" : "") + "></web-forms>";
    }

    // Overload
    public String exportToWebFormsTag(int width, int height, String src) {
        return exportToWebFormsTag(width + "px", height + "px", src);
    }

    // Helper Methods
    private String getNameByIndex(int index) {
        List<String> keys = new ArrayList<>(webFormsData.keySet());
        return (index >= 0 && index < keys.size()) ? keys.get(index) : "";
    }

    private void changeNameByIndex(int index, String newName) {
        List<String> keys = new ArrayList<>(webFormsData.keySet());
        if (index >= 0 && index < keys.size()) {
            String value = webFormsData.remove(keys.get(index));
            webFormsData.put(newName, value);
        }
    }

    private String removeOuter(String text, String startString, String endString) {
        int start = text.indexOf(startString);
        if (start == -1) return text;

        int end = text.indexOf(endString, start);
        if (end == -1) return text;

        int lengthToRemove = (end - start) + endString.length();
        return text.substring(0, start) + text.substring(end + endString.length());
    }
    
    // InputPlace Class
    public static class InputPlace {
        public static String id(String id) {
            return id;
        }

        public static String name(String name) {
            return '(' + name + ')';
        }

        public static String name(String name, int index) {
            return '(' + name + ')' + index;
        }

        public static String tag(String tag) {
            return '<' + tag + '>';
        }

        public static String tag(String tag, int index) {
            return '<' + tag + '>' + index;
        }

        public static String className(String className) {
            return '{' + className + '}';
        }

        public static String className(String className, int index) {
            return '{' + className + '}' + index;
        }

        public static String query(String query) {
            return "*" + query.replace("=", "$[eq];");
        }

        public static String queryAll(String query) {
            return "[" + query.replace("=", "$[eq];");
        }
    }

    // ExtensionWebFormsMethods Class
    public static class ExtensionWebFormsMethods {
        public static String appendPlace(String text, String value) {
            if (text.length() < 1) {
                return value;
            }

            if (text.charAt(0) != '>') {
                text = '>' + text;
            }

            return text + "|" + value;
        }

        public static String exportToWebFormsTag(String src) {
            return "<web-forms src=\"" + src + "\"></web-forms>";
        }

        // Overload
        public static String exportToWebFormsTag(String src, int width, int height) {
            return "<web-forms src=\"" + src + "\" width=\"" + width + "\" height=\"" + height + "\"></web-forms>";
        }

        public static String exportActionControlsToWebFormsTag(String actionControls) {
            return "<web-forms ac=\"" + actionControls + "\"></web-forms>";
        }

        public static String removeOuter(String text, String startString, String endString) {
            int start = text.indexOf(startString);
            if (start == -1) return text;

            int end = text.indexOf(endString, start);
            if (end == -1) return text;

            int lengthToRemove = (end - start) + endString.length();
            return text.substring(0, start) + text.substring(end + endString.length());
        }
    }
}

File: swift/WebForms.swift
import Foundation

class WebForms {
    private var webFormsData = [String: String]()

    // Add
    func addId(_ inputPlace: String, _ id: String) {
        webFormsData["ai\(inputPlace)"] = id
    }
    
    func addName(_ inputPlace: String, _ name: String) {
        webFormsData["an\(inputPlace)"] = name
    }
    
    func addValue(_ inputPlace: String, _ value: String) {
        webFormsData["av\(inputPlace)"] = value
    }
    
    func addClass(_ inputPlace: String, _ className: String) {
        webFormsData["ac\(inputPlace)"] = className
    }
    
    func addStyle(_ inputPlace: String, _ style: String) {
        webFormsData["as\(inputPlace)"] = style
    }
    
    func addOptionTag(_ inputPlace: String, text: String, value: String, selected: Bool = false) {
        webFormsData["ao\(inputPlace)"] = "\(value)|\(text)\(selected ? "|1" : "")"
    }
    
    func addCheckBoxTag(_ inputPlace: String, text: String, value: String, checked: Bool = false) {
        webFormsData["ak\(inputPlace)"] = "\(value)|\(text)\(checked ? "|1" : "")"
    }
    
    func addTitle(_ inputPlace: String, _ title: String) {
        webFormsData["al\(inputPlace)"] = title
    }
    
    func addText(_ inputPlace: String, _ text: String) {
        webFormsData["at\(inputPlace)"] = text.replacingOccurrences(of: "\n", with: "$[ln];")
    }
    
    func addAttribute(_ inputPlace: String, _ attribute: String, _ value: String = "") {
        webFormsData["aa\(inputPlace)"] = "\(attribute)|\(value)"
    }
    
    func addTag(_ inputPlace: String, tagName: String, id: String = "") {
        webFormsData["nt\(inputPlace)"] = "\(tagName)\(id.isEmpty ? "" : "|\(id)")"
    }

    // Set
    func setId(_ inputPlace: String, _ id: String) {
        webFormsData["si\(inputPlace)"] = id
    }
    
    func setName(_ inputPlace: String, _ name: String) {
        webFormsData["sn\(inputPlace)"] = name
    }
    
    func setValue(_ inputPlace: String, _ value: String) {
        webFormsData["sv\(inputPlace)"] = value
    }
    
    func setClass(_ inputPlace: String, _ className: String) {
        webFormsData["sc\(inputPlace)"] = className
    }
    
    func setStyle(_ inputPlace: String, _ style: String) {
        webFormsData["ss\(inputPlace)"] = style
    }
    
    func setOptionTag(_ inputPlace: String, text: String, value: String, selected: Bool = false) {
        webFormsData["so\(inputPlace)"] = "\(value)|\(text)\(selected ? "|1" : "")"
    }
    
    func setChecked(_ inputPlace: String, _ checked: Bool = false) {
        webFormsData["sk\(inputPlace)"] = checked ? "1" : "0"
    }
    
    func setCheckBoxTagToList(_ inputPlace: String, text: String, value: String, checked: Bool = false) {
        webFormsData["sk\(inputPlace)"] = "\(value)|\(text)\(checked ? "|1" : "")"
    }

    func setTitle(_ inputPlace: String, _ title: String) {
        webFormsData["sl\(inputPlace)"] = title
    }

    func setText(_ inputPlace: String, _ text: String) {
        webFormsData["st\(inputPlace)"] = text.replacingOccurrences(of: "\n", with: "$[ln];")
    }

    func setAttribute(_ inputPlace: String, _ attribute: String, _ value: String = "") {
        webFormsData["sa\(inputPlace)"] = "\(attribute)\(value.isEmpty ? "" : "|\(value)")"
    }

    func setWidth(_ inputPlace: String, _ width: String) {
        webFormsData["sw\(inputPlace)"] = width
    }

    func setWidth(_ inputPlace: String, _ width: Int) {
        setWidth(inputPlace, "\(width)px")
    }

    func setHeight(_ inputPlace: String, _ height: String) {
        webFormsData["sh\(inputPlace)"] = height
    }

    func setHeight(_ inputPlace: String, _ height: Int) {
        setHeight(inputPlace, "\(height)px")
    }

    // Insert
    func insertId(_ inputPlace: String, _ id: String) {
        webFormsData["ii\(inputPlace)"] = id
    }
    
    func insertName(_ inputPlace: String, _ name: String) {
        webFormsData["in\(inputPlace)"] = name
    }
    
    func insertValue(_ inputPlace: String, _ value: String) {
        webFormsData["iv\(inputPlace)"] = value
    }
    
    func insertClass(_ inputPlace: String, _ className: String) {
        webFormsData["ic\(inputPlace)"] = className
    }
    
    func insertStyle(_ inputPlace: String, _ style: String) {
        webFormsData["is\(inputPlace)"] = style
    }
    
    func insertOptionTag(_ inputPlace: String, text: String, value: String, selected: Bool = false) {
        webFormsData["io\(inputPlace)"] = "\(value)|\(text)\(selected ? "|1" : "")"
    }
    
    func insertCheckBoxTag(_ inputPlace: String, text: String, value: String, checked: Bool = false) {
        webFormsData["ik\(inputPlace)"] = "\(value)|\(text)\(checked ? "|1" : "")"
    }
    
    func insertTitle(_ inputPlace: String, _ title: String) {
        webFormsData["il\(inputPlace)"] = title
    }
    
    func insertText(_ inputPlace: String, _ text: String) {
        webFormsData["it\(inputPlace)"] = text.replacingOccurrences(of: "\n", with: "$[ln];")
    }
    
    func insertAttribute(_ inputPlace: String, _ attribute: String, _ value: String = "") {
        webFormsData["ia\(inputPlace)"] = "\(attribute)\(value.isEmpty ? "" : "|\(value)")"
    }

    // Delete
    func deleteId(_ inputPlace: String) {
        webFormsData["di\(inputPlace)"] = "1"
    }
    
    func deleteName(_ inputPlace: String) {
        webFormsData["dn\(inputPlace)"] = "1"
    }
    
    func deleteValue(_ inputPlace: String) {
        webFormsData["dv\(inputPlace)"] = "1"
    }
    
    func deleteClass(_ inputPlace: String, _ className: String) {
        webFormsData["dc\(inputPlace)"] = className
    }
    
    func deleteStyle(_ inputPlace: String, _ styleName: String) {
        webFormsData["ds\(inputPlace)"] = styleName
    }
    
    func deleteOptionTag(_ inputPlace: String, _ value: String) {
        webFormsData["do\(inputPlace)"] = value
    }
    
    func deleteCheckBoxTag(_ inputPlace: String, _ value: String) {
        webFormsData["dk\(inputPlace)"] = value
    }
    
    func deleteTitle(_ inputPlace: String) {
        webFormsData["dl\(inputPlace)"] = "1"
    }
    
    func deleteText(_ inputPlace: String) {
        webFormsData["dt\(inputPlace)"] = "1"
    }
    
    func deleteAttribute(_ inputPlace: String, _ attribute: String) {
        webFormsData["da\(inputPlace)"] = attribute
    }
    
    func delete(_ inputPlace: String) {
        webFormsData["de\(inputPlace)"] = "1"
    }

    // Other
    func setBackgroundColor(_ inputPlace: String, _ color: String) {
        webFormsData["bc\(inputPlace)"] = color
    }
    
    func setTextColor(_ inputPlace: String, _ color: String) {
        webFormsData["tc\(inputPlace)"] = color
    }
    
    func setFontName(_ inputPlace: String, _ name: String) {
        webFormsData["fn\(inputPlace)"] = name
    }
    
    func setFontSize(_ inputPlace: String, _ size: String) {
        webFormsData["fs\(inputPlace)"] = size
    }
    
    func setFontSize(_ inputPlace: String, _ size: Int) {
        setFontSize(inputPlace, "\(size)px")
    }
    
    func setFontBold(_ inputPlace: String, _ bold: Bool) {
        webFormsData["fb\(inputPlace)"] = bold ? "1" : "0"
    }
    
    func setVisible(_ inputPlace: String, _ visible: Bool) {
        webFormsData["vi\(inputPlace)"] = visible ? "1" : "0"
    }
    
    func setTextAlign(_ inputPlace: String, _ align: String) {
        webFormsData["ta\(inputPlace)"] = align
    }
    
    func setReadOnly(_ inputPlace: String, _ readOnly: Bool) {
        webFormsData["sr\(inputPlace)"] = readOnly ? "1" : "0"
    }
    
    func setDisabled(_ inputPlace: String, _ disabled: Bool) {
        webFormsData["sd\(inputPlace)"] = disabled ? "1" : "0"
    }
    
    func setMinLength(_ inputPlace: String, _ length: Int) {
        webFormsData["mn\(inputPlace)"] = "\(length)"
    }
    
    func setMaxLength(_ inputPlace: String, _ length: Int) {
        webFormsData["mx\(inputPlace)"] = "\(length)"
    }
    
    func setSelectedValue(_ inputPlace: String, _ value: String) {
        webFormsData["ts\(inputPlace)"] = value
    }
    
    func setSelectedIndex(_ inputPlace: String, _ index: Int) {
        webFormsData["ti\(inputPlace)"] = "\(index)"
    }
    
    func setCheckedValue(_ inputPlace: String, _ value: String, _ selected: Bool) {
        webFormsData["ks\(inputPlace)"] = "\(value)|\(selected ? "1" : "0")"
    }
    
    func setCheckedIndex(_ inputPlace: String, _ index: Int, _ selected: Bool) {
        webFormsData["ki\(inputPlace)"] = "\(index)|\(selected ? "1" : "0")"
    }
    
    func callScript(_ scriptText: String) {
        webFormsData["_"] = scriptText.replacingOccurrences(of: "\n", with: "$[ln];")
    }
    
    func loadUrl(_ inputPlace: String, _ url: String) {
        webFormsData["lu\(inputPlace)"] = url
    }

    // Increase
    func increaseMinLength(_ inputPlace: String, _ value: Int) {
        webFormsData["+n\(inputPlace)"] = "\(value)"
    }
    
    func increaseMaxLength(_ inputPlace: String, _ value: Int) {
        webFormsData["+x\(inputPlace)"] = "\(value)"
    }
    
    func increaseFontSize(_ inputPlace: String, _ value: Int) {
        webFormsData["+f\(inputPlace)"] = "\(value)"
    }
    
    func increaseWidth(_ inputPlace: String, _ value: Int) {
        webFormsData["+w\(inputPlace)"] = "\(value)"
    }
    
    func increaseHeight(_ inputPlace: String, _ value: Int) {
        webFormsData["+h\(inputPlace)"] = "\(value)"
    }
    
    func increaseValue(_ inputPlace: String, _ value: Int) {
        webFormsData["+v\(inputPlace)"] = "\(value)"
    }

    // Decrease
    func decreaseMinLength(_ inputPlace: String, _ value: Int) {
        webFormsData["-n\(inputPlace)"] = "\(value)"
    }
    
    func decreaseMaxLength(_ inputPlace: String, _ value: Int) {
        webFormsData["-x\(inputPlace)"] = "\(value)"
    }
    
    func decreaseFontSize(_ inputPlace: String, _ value: Int) {
        webFormsData["-f\(inputPlace)"] = "\(value)"
    }
    
    func decreaseWidth(_ inputPlace: String, _ value: Int) {
        webFormsData["-w\(inputPlace)"] = "\(value)"
    }
    
    func decreaseHeight(_ inputPlace: String, _ value: Int) {
        webFormsData["-h\(inputPlace)"] = "\(value)"
    }
    
    func decreaseValue(_ inputPlace: String, _ value: Int) {
        webFormsData["-v\(inputPlace)"] = "\(value)"
    }

    // Pre Runner
    func assignDelay(_ second: Float, index: Int = -1) {
        guard let currentName = getNameByIndex(index) else { return }
        webFormsData[changeName(currentName, with: ":\(second)", index: index)] = currentName
    }

    func assignDelayChange(_ second: Float, index: Int = -1) {
        guard let currentName = getNameByIndex(index) else { return }
        let updatedName = currentName.removeOuter(":", ")")
        webFormsData[changeName(updatedName, with: ":\(second)", index: index)] = updatedName
    }

    func assignInterval(_ second: Float, index: Int = -1) {
        guard let currentName = getNameByIndex(index) else { return }
        webFormsData[changeName(currentName, with: "(\(second)", index: index)] = currentName
    }

    func assignIntervalChange(_ second: Float, index: Int = -1) {
        guard let currentName = getNameByIndex(index) else { return }
        let updatedName = currentName.removeOuter("(", ")")
        webFormsData[changeName(updatedName, with: "(\(second)", index: index)] = updatedName
    }

    // Get
    func getFormsActionData() -> String {
        return webFormsData.map { "\($0.key)=\($0.value)" }.joined(separator: "\n")
    }

    func response() -> String {
        return "[web-forms]\(getFormsActionData())"
    }

    func getFormsActionDataLineBreak() -> String {
        let entries = webFormsData.map { "\($0.key)=\($0.value.replacingOccurrences(of: "\"", with: "$[dq];"))" }
        let lastIndex = entries.count - 1
        return entries.enumerated().map { $0.element + ($0.offset < lastIndex ? "$[sln];" : "") }.joined(separator: "")
    }

    // Export
    func exportToWebFormsTag(src: String? = nil) -> String {
        return "<web-forms ac=\"\(getFormsActionDataLineBreak())\" \(src != nil ? "src=\"\(src!)\"" : "")></web-forms>"
    }

    // Overload
    func exportToWebFormsTag(width: String, height: String, src: String? = nil) -> String {
        return "<web-forms ac=\"\(getFormsActionDataLineBreak())\" width=\"\(width)\" height=\"\(height)\" \(src != nil ? "src=\"\(src!)\"" : "")></web-forms>"
    }

    // Overload
    func exportToWebFormsTag(width: Int, height: Int, src: String? = nil) -> String {
        return exportToWebFormsTag(width: "\(width)px", height: "\(height)px", src: src)
    }
    
    // Helper Methods
    private func getNameByIndex(_ index: Int) -> String? {
        // Implement logic to fetch name by index
        // This method is a placeholder, logic needs to be defined upon full context
        return nil
    }
    
    private func changeName(_ currentName: String, with prefix: String, index: Int) -> String {
        // Implement logic to change name depending on current name and index
        // This method is a placeholder, logic needs to be defined upon full context
        return currentName
    }
}

struct InputPlace {
    static func id(_ id: String) -> String {
        return id
    }
    
    static func name(_ name: String) -> String {
        return "(\(name))"
    }
    
    static func name(_ name: String, index: Int) -> String {
        return "(\(name))\(index)"
    }
    
    static func tag(_ tag: String) -> String {
        return "<\(tag)>"
    }
    
    static func tag(_ tag: String, index: Int) -> String {
        return "<\(tag)>\(index)"
    }
    
    static func className(_ className: String) -> String {
        return "{\(className)}"
    }
    
    static func className(_ className: String, index: Int) -> String {
        return "{\(className)}\(index)"
    }
    
    static func query(_ query: String) -> String {
        return "*\(query.replacingOccurrences(of: "=", with: "$[eq];"))"
    }
    
    static func queryAll(_ query: String) -> String {
        return "[\(query.replacingOccurrences(of: "=", with: "$[eq];"))"
    }
}

extension String {
    func appendPlace(_ value: String) -> String {
        if count < 1 {
            return value
        }
        
        var updatedText = self
        
        if updatedText[updatedText.startIndex] != ">" {
            updatedText = ">" + updatedText
        }
        
        return updatedText + "|" + value
    }
    
    func exportToWebFormsTag() -> String {
        return "<web-forms src=\"\(self)\"></web-forms>"
    }
    
    // Overload
    func exportToWebFormsTag(width: Int, height: Int) -> String {
        return "<web-forms src=\"\(self)\" width=\"\(width)\" height=\"\(height)\"></web-forms>"
    }
    
    func exportActionControlsToWebFormsTag(actionControls: String) -> String {
        return "<web-forms ac=\"\(actionControls)\"></web-forms>"
    }
    
    func removeOuter(startString: String, endString: String) -> String {
        guard let start = self.range(of: startString)?.lowerBound else {
            return self
        }
        
        guard let end = self.range(of: endString, range: start..<self.endIndex)?.upperBound else {
            return self
        }
        
        var result = self
        result.removeSubrange(start..<end)
        return result
    }
}